spec/activerecord-bitemporal/bitemporal_spec.rb

```diff
@@ -2702,6 +2702,48 @@ class EmployeeWithUniquness < Employee
     end
   end
 
+  describe ".freeze_transaction_datetime" do
+    context 'with freeze_transaction_datetime' do
+      it 'same transaction_time for create, update and delete operations' do
+        employee_for_update = Employee.create!(name: "Jone")
+        employee_for_destroy = Employee.create!(name: "Mami")
+
+        ActiveRecord::Bitemporal.freeze_transaction_datetime do
+          employee_for_create = Employee.create!(name: "Homu")
+          employee_for_update.update!(name: "Bonnie")
+          employee_for_destroy.destroy!
+
+          create_time = employee_for_create.reload.transaction_from
+          update_time = employee_for_update.reload.transaction_from
+          destroy_time = Employee.bitemporal_for(employee_for_destroy.id).ignore_bitemporal_datetime.maximum(:transaction_from)
+
+          expect([create_time,
+                  update_time,
+                  destroy_time].uniq.size).to eq 1
```

Comment:
Intuitively, transaction time with zero duration is nonsense because it cannot be referenced in queries.
Do we need validations to prevent this? (e.g. disallow `transaction_from == transaction_to`)

Reply:
fixed 88c5315
added validation for transaction_datetime.